src/NServiceBus.Persistence.AzureStorage/Timeout/TimeoutLogic/TimeoutPersister.cs

```diff
@@ -133,12 +133,17 @@ public TimeoutPersister(string timeoutConnectionString, string timeoutDataTableN
 
             var results = await timeoutDataTable.ExecuteQueryAsync(query, take: 1000).ConfigureAwait(false);
 
+            var deletionTasks = new List<Task>(4);
             foreach (var timeoutDataEntityBySaga in results)
             {
-                await DeleteState(timeoutDataEntityBySaga.StateAddress).ConfigureAwait(false);
-                await DeleteTimeEntity(timeoutDataTable, timeoutDataEntityBySaga.Time.ToString(partitionKeyScope), timeoutDataEntityBySaga.RowKey).ConfigureAwait(false);
-                await DeleteMainEntity(timeoutDataTable, timeoutDataEntityBySaga.RowKey, string.Empty).ConfigureAwait(false);
-                await DeleteSagaEntity(timeoutDataTable, timeoutDataEntityBySaga).ConfigureAwait(false);
+                deletionTasks.Add(DeleteState(timeoutDataEntityBySaga.StateAddress));
```

Comment:
@boblangley @dbelcham same thoughts as in my previous comment Are you OK with these changes?

Reply:
I don't see any harm, and the tests are passing :shipit: